### PR TITLE
 - Added option to only install dependencies for packages where only a virtualenv containing dependencies is required

### PR DESCRIPTION
**Pull request by phillipod**
I work with an environment where the virtualenv only contains dependencies for the application, the actual application code is deployed separately. 

This change allows the user to specify --dependencies-only - this results in no action being taken by deploy.install_package()

### CODE REVIEW

**Comment by nailor** [comment]
Thank you for your pull request!

The idea of this flag is to skip running `pip install .` on a project, right? Maybe it should be called `--skip-install` or something like that. It also needs documentation. The documentation should also state that this flag won't respect whatever is defined in the `setup.py`.

The last part might be a bit cumbersome. Say user wants to package a third party tool that defines all its dependencies in `setup.py`. Is the user expected to copy those over to `requirements.txt`? Requires some careful wording in the documentation.

Maybe we could run something like `pip install -e .` instead? Though I can foresee that it leads to a lot of renaming hassles when we move the virtual environment in place.

tl;dr Just make a good documentation and change the name of the flag and I'm willing to merge this as it is for now :)

**Comment by jhermann** [comment]
The purpose is to just deploy a bunch of PyPI packages I guess (e.g. deploying a well-defined runtime environment for a bunch of scripts), so you wouldn't have a setup.py at all.  Docs should include a practical and typical use-case.

BTW, the `is not True` in there is mind-bending, `if not self.skip_install` is much more idiomatic.

**Comment by nailor** [comment]
Yeah, one could work around this by just tossing the `setup.py` out of the window. Problem then is that the actual installation will fail: `pip install .` will exit with 1 and we use `subprocess.check_call` around it.

I'd still prefer an explicit skip flag. If we were to make an implicit detection of existence of the `setup.py` and avoid running `pip install .` when it is missing, we would probably create a pile of builds that pass but don't really reflect what the user want.

**Comment by phillipod** [comment]
Thanks for the feedback! Yep, I'd tend to expect this to be used where there is no setup.py - actually, I had considered creating a 'setup.py' script that just did nothing, but boilerplate makes me sad when it's in my build systems.

I've added documentation for the feature and attempted to make it clear that it's the users responsibility to deal with whatever may have been in an existing setup.py, if such did exist. 

I've renamed the parameter '--skip-install', however it does seem not to follow the existing convention such as that for '--no-test' - both in name and function. It seems to me that 'python setup.py test' is only called if 'setup.py' exists. Let me know if you'd like me to adjust this.

Cheers,
Phillip

**Comment by nailor** [comment]
I think the `--no-test` should be named to `--skip-test` in future leaving the old command for backwards compatibility. Not too fond of the flag name I came up with :p 

I do agree that it is sort of weird that the test command tests for existence of the setup file, maybe that should be removed too.

**Comment by nailor** [comment]
Btw, I like how you provided a concrete example in the docs and thanks for your pull request!

**Comment by phillipod** [comment]
No problem at all, and thank you for a tool that's going to make my life much easier!
